src/btree/bt_walk.c

```diff
@@ -65,14 +65,25 @@ __ref_index_slot(WT_SESSION_IMPL *session,
 				goto found;
 			}
 		}
+		/*
+		 * We failed to get the page index and slot reference,
+		 * yield before retrying, and if we've yielded enough times,
+		 * start sleeping so we don't burn CPU to no purpose.
+		 */
+		if (++yield_count < WT_THOUSAND) {
```

Comment:
yield_count increment can be moved inside the if condition block, so that when yield_count is above 1000, it doesnt unnecessarily increment.